internal/buffer/loc.go

```diff
@@ -9,6 +9,16 @@ type Loc struct {
 	X, Y int
 }
 
+// LocVoid returns a Loc strictly smaller then any valid buffer location
+func LocVoid() Loc {
+	return Loc{-1, -1}
+}
+
+// IsVoid returns true if the location l is void
+func (l Loc) IsVoid() bool {
+	return l == LocVoid()
+}
+
```

Comment:
WTF is this, sorry.

I was about to suggest something like:

```
const InvalidLoc = Loc{-1, -1}
```

but then I recalled that Go doesn't support constant structs.

So I think we should just keep using directly `Loc{-1, -1}` (and explicit checks like `loc == Loc{-1, -1}`, without helpers), there's nothing terrible about it.

Reply:
I also thought first about constant structs.

The reason that I made this change in this PR is that the "internal" value `Loc{-1, -1}` is now exposed to Lua scripts: If a submatch is not filled in a match, then we need a way to indicate that. An example would be searching for "a([xy])|b([uv])" in "ax". The first submatch would be "x" and the second one would be void. (In Go, the indices of a void submatch are `-1`.) I thought that something like `loc:IsVoid()` looks cleaner in in a Lua script.

Does this convince you, or do you still want me to remove `LocVoid` and `IsVoid`?

Reply:
First, the use of the word "void" here is very confusing, isn't it? Why "void"? (Now I understand it refers to this specific use case of "a submatch is not filled in a match", but how is a casual person supposed to guess that, and why limit the API to this narrow use case?)

What about just:

```
func (l Loc) IsValid() bool {
	return l.X >= 0 && l.Y >= 0
}
```

?